core-bundle/src/Resources/contao/elements/ContentGallery.php

```diff
@@ -258,10 +258,12 @@ protected function compile()
 				// Image / empty cell
 				if (($j + $i) < $limit && null !== ($image = $images[$i + $j] ?? null))
 				{
-					$cellData = $figureBuilder
+					$figure = $figureBuilder
 						->fromFilesModel($image)
-						->build()
-						->getLegacyTemplateData($this->imagemargin);
+						->build();
+
+					$cellData = $figure->getLegacyTemplateData($this->imagemargin);
+					$cellData['figure'] = $this;
```

Comment:
```suggestion
					$cellData['figure'] = $figure;
```

Shouldn't this be `$figure` instead of `$this`?